contracts/Everlong.sol

```diff
@@ -399,22 +462,59 @@ contract Everlong is IEverlong {
 
     // TODO: Decide if we want to put a slippage guard here.
     //
+    /// @notice Closes mature positions in the Everlong portfolio.
+    /// @param _limit Limit the maximum number of positions to close.
+    ///               - Zero indicates no limit.
+    /// @return output Amount of assets received from the closed positions.
+    function closeMaturedPositions(
+        uint256 _limit
+    ) external returns (uint256 output) {
+        output = _closeMaturedPositions(_limit);
+    }
+
     /// @dev Close only matured positions in the portfolio.
+    /// @param _limit The maximum number of positions to close.
+    ///               A value of zero indicates no limit.
     /// @return output Proceeds of closing the matured positions.
-    function _closeMaturedPositions() internal returns (uint256 output) {
+    function _closeMaturedPositions(
+        uint256 _limit
+    ) internal returns (uint256 output) {
+        // A value of zero for `_limit` indicates no limit.
+        if (_limit == 0) {
+            _limit = type(uint256).max;
+        }
+
+        // Iterate through positions from most to least mature.
+        // Exit if:
+        // - There are no more positions.
+        // - The current position is not mature.
+        // - The limit on closed positions has been reached.
         IEverlong.Position memory position;
-        while (!_portfolio.isEmpty()) {
+        uint256 count = 0;
+        while (!_portfolio.isEmpty() && count < _limit) {
```

Comment:
While loops are fine, but you could also make this a for-loop. It has all of the necessary ingredients